### PR TITLE
EPMDJ-5963 [BE] For an Offline Service in a Service group should not start a test session

### PR DESCRIPTION
**Pull request by Mazel-Tovr**

### CODE REVIEW

File: core/src/main/kotlin/endpoints/AgentManager.kt [hunk 1]
```diff
@@ -220,6 +220,7 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
         }.getOrDefault(id, persistentHashMapOf())
         if (instances.isEmpty()) {
             logger.info { "Agent with id '${id}' was disconnected" }
+            entryOrNull(id)?.updateAgent { it.copy(status = AgentStatus.OFFLINE) }?.persistToDatabase()
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/endpoints/AgentManager.kt]: why here do we save status in db?

File: tests/src/test/kotlin/e2e/plugin/PluginTest.kt [hunk 2]
```diff
@@ -63,7 +63,7 @@ class PluginTest : E2EPluginTest() {
                 println("hi ag3")
             }
             uiWatcher { channel ->
-                waitForMultipleAgents(channel)
+                waitForMultipleAgents(channel, 3)
```
Comment (LikhanovMaxim) [inline tests/src/test/kotlin/e2e/plugin/PluginTest.kt]: move to default value

**Review by LikhanovMaxim — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/endpoints/AgentManager.kt (hunk 1 above)
Comment (Mazel-Tovr) [reply core/src/main/kotlin/endpoints/AgentManager.kt]: Before the changes, we saved all agent statuses changes in the database, but now, we will store only 2 states in the database, either the agent is not registered, or offline , if you do not save the status to offline, then after restarting the system, the agent will rise with the status not registered. 
Although, maybe it's better to save the agent's status offline immediately after registering the agent ?

File: tests/src/test/kotlin/e2e/plugin/PluginTest.kt [hunk 3]
```diff
@@ -82,11 +83,11 @@ class PluginTest : E2EPluginTest() {
 
     }
 
-    private suspend fun waitForMultipleAgents(channel: Channel<GroupedAgentsDto>) {
+    private suspend fun waitForMultipleAgents(channel: Channel<GroupedAgentsDto>, instanceCount: Int) {
```
Comment (LikhanovMaxim) [inline tests/src/test/kotlin/e2e/plugin/PluginTest.kt]:
```suggestion
    private suspend fun waitForMultipleAgents(channel: Channel<GroupedAgentsDto>, instanceCount: Int = 3) {
```

File: core/src/main/kotlin/endpoints/AgentManager.kt [hunk 4]
```diff
@@ -209,6 +209,7 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
         adminData(agentId).updateSettings(dto.systemSettings)
         pluginsToAdd.forEach { plugin -> ensurePluginInstance(entry, plugin) }
         info.sync()
+        info.copy(status = AgentStatus.OFFLINE).persistToDatabase()
```
Comment (LikhanovMaxim) [inline core/src/main/kotlin/endpoints/AgentManager.kt]: move to another place: in starting admin

**Comment by Mazel-Tovr** [comment]
fixed in https://github.com/Drill4J/admin/pull/153

**Pull request closed without merging**